internal/cli/cmd/cluster/buildx.go

```diff
@@ -67,34 +92,40 @@ func newSetupBuildxCmd(cmdName string) *cobra.Command {
 			return rpcerrors.Errorf(codes.Internal, "no builders available")
 		}
 
-		state, err := ensureStateDir(*stateDir, "buildkit", "proxy")
+		state, err := ensureStateDir(*stateDir, buildkitProxyPath)
 		if err != nil {
 			return err
 		}
 		fmt.Fprintf(console.Debug(ctx), "Using state path %q\n", state)
 
-		var md buildxMetadata
+		if proxyAlreadyExists(state) {
+			console.SetStickyContent(ctx, "build", existingProxyMessage(*stateDir))
+			return nil
+		}
+
+		md := buildxMetadata{
+			NodeGroupName: *name,
+		}
 		for _, p := range available {
 			sockPath := filepath.Join(state, fmt.Sprintf("%s.sock", p))
 			md.Instances = append(md.Instances, buildxInstanceMetadata{
-				Platform:   p,
-				SocketPath: sockPath,
+				Platform:           p,
+				SocketPath:         sockPath,
+				BackgroundProxyPID: -1, // -1 means proxy is not in background
```

Comment:
pid 0 is also not defined so I'd lean on using that as the marker of pid not being set.